src/CropManager.sol

```diff
@@ -125,17 +152,14 @@ contract CropManagerImp {
         CropLike(crop).flee(urp, msg.sender);
     }
 
-    function frob(address crop, address u, address v, address w, int256 dink, int256 dart) external {
-        require(u == msg.sender && v == msg.sender && w == msg.sender, "CropManager/not-allowed");
-
-        // Note: This simplification only works because of the u == v == msg.sender restriction above
+    function frob(address crop, address u, address v, address w, int256 dink, int256 dart) external allowed(u) {
+        require(u == v && w == msg.sender, "CropManager/not-matching");
```

Comment:
I'm wondering if maybe we should change the `w == msg.sender` requirement to something else... Seems a bit inconvenient to allow management of the Vault by a third party, but that third party can only draw/wipe DAI to/from their own address. It's not obvious what the better replacement is, though. I guess for a DeFi Saver-type use case, `w == msg.sender` actually works quite well, and the third party can always then call `Vat.move` directly. So, I guess I've talked myself out of changing it, but I'll send this comment anyway in case it inspires anyone else.

Reply:
I thought about the `u == v` requirement too--in the managed case, you can get collateral into the Vault by `fluxing` appropriately first, but there's no way to get the collateral out and put it elsewhere for the allowed address. Maybe that's fine too--arguably it has a nice safety property. But I'm wondering if some applications built on top will need stronger permissions.

Reply:
@kmbarry1 hmmm not sure if I understanding the last one but you can take the collateral from a third party vault that you manage using `frob + flux`. `flux` also has the managed permission.

Reply:
Ah true. You just can't do it directly in the `frob` (so can add gas costs/complexity). I think this is fine for now, we'll just have to see what sort of feedback we get.